xrviz/coord_setter.py

```diff
@@ -18,7 +28,7 @@ def __init__(self, data):
 
     def set_coords(self, data):
         """
-        to set the coords
+        To set the coords
```

Comment:
Called when the data attribute of the interface acquires new coordinates

Why is this ever called at all? In this pane, we *define* the coordinates, so this may lead to recursion.

Reply:
The actual set/reset of coordinates happens in `dashboard.py`. Here `self.coord_selector.value = list(self.data.coords)` ensures that the default coords are also present by interpreting from the data, in case they  have been reset. Here just updated values are being displayed.

Reply:
I disagree: at `self.coord_selector = ` in `__init__`, you set the starting values. Now whenever `self.coord_selector.value` changes, because of user action, the outer class takes care to set the coordinates on the xarray object, but why do you need to set the `.value` again? I believe this will always be a no-op (but you could set a breakpoint or log here, if you don't think so).

Reply:
I have tried removing `self.coord_selector.value = list(self.data.coords)`. If the user removes `time` or `sigma`, these(indexed coord) would not be available in the values in `Set Coords` but would be present as coords in data and will be displayed as coords in `Axes`. This would be inconsistent.

> why do you need to set the `.value` again?

To get back the values of indexed coordinates in case they have been reset by the user. (The user can select and try resetting an indexed coord but it would again be added to the values on right with the help of this line).

I have added `logger.debug(self.coord_selector.value)` before setting `.value`.

Reply:
> I believe this will always be a no-op.

Yes, it is no-op but only for the case when the value that has been set/reset is not indexed coord.